src/App.tsx

```diff
@@ -15,47 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type RecorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: RecorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReverse] = React.useState(false);
+  const [sortType, setSortType] = React.useState(SortType.NONE);
+
+  const sortByAlphabeth = () => setSortType(SortType.ALPHABET);
+  const sortByLength = () => setSortType(SortType.LENGTH);
+  const reverse = () => setReverse(!isReversed);
+  const reset = () => {
+    setSortType(SortType.NONE);
+    setReverse(false);
+  };
+
+  const goodsList = getReorderedGoods(
+    goodsFromServer,
+    { sortType, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button is-ingo', {
+            'is-light': sortType !== SortType.ALPHABET,
+          })}
+          onClick={sortByAlphabeth}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames('button is-success', {
+            'is-light': sortType !== SortType.LENGTH,
+          })}
+          onClick={sortByLength}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={reverse}
         >
           Reverse
         </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(isReversed || sortType !== SortType.NONE)
+        && (
```

Comment:
move this to the separate variable
```suggestion
        {... && (
```